### ISSUE TITLE
False Positive Error with 'this' and object literal shorthand

### ISSUE THREAD

**Issue by jbcpollak** [eslint/eslint#6013]
**What version of ESLint are you using?**

2.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Node v5.7.1, server side code, not using Babel

**Please show your full configuration:**

```
// Based on the awesome config from walmart labs:
//https://github.com/walmartlabs/eslint-config-defaults
{
	"extends": "defaults/configurations/google",
	"rules": {
		"max-len": [2, 110, 2],
		"indent": [2, "tab"],
		"dot-location": [2, "property"],
		"no-mixed-spaces-and-tabs": [2, "smart-tabs"],
		"no-console": [2],
		"no-var": [2]
	},
	"env": {
		"node": true,
		"es6": true,
		"mocha": true
	},
    "ecmaFeatures": {
        "modules": true,
        "classes": true,
        "arrowFunctions": true,
        "blockBindings": true,
        "templateStrings": true
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```
    myFunc() {

		let that = 1;

		log.debug({
			that,
			this
		}, 'log message here');
    }
```

**What did you expect to happen?**

No error should be returned

**What actually happened? Please include the actual, raw output from ESLint.**

```
  52:4  error  'this' is not defined  no-undef
```

This code works as expected:

```
    myFunc() {

		let that = 1;

		log.debug({
			that,
			this: this
		}, 'log message here');
    }
```

Probably a rare usecase, but in this case we wanted to log 'this' in our logger and have it appear as 'this'.

I think this is a regression, as I _think_ this code has been in place for a while without throwing an error and with the same eslint config. We've just upgraded eslint.

**Comment by mysticatea** [comment]
Thank you for this issue.

```js
let foo = {this};
```

This is a syntax error.
If you use shorthand properties, you cannot use keywords there.

https://tc39.github.io/ecma262/2016/#prod-ObjectLiteral

PropertyDefinition -> IdentifierReference -> Identifier but not ReserevedWord

I will open an issue on Acorn later.

**Comment by jbcpollak** [comment]
Interesting. I see it doesn't work in Chrome - either Node accepted it or I never hit that log condition, which seems unlikely. Thanks.

**Comment by mysticatea** [comment]
This has been fixed in Acorn.
I'm waiting for the next release.

**Comment by mysticatea** [comment]
Now Acorn v3.2.0 has been released: https://github.com/ternjs/acorn/releases/tag/3.2.0
